src/app/mod.rs

```diff
@@ -1227,6 +1230,32 @@ impl<'a, 'b> App<'a, 'b> {
         self.get_matches_from_safe(&mut env::args_os())
     }
 
+    /// Similar to [`App::get_matches`] but also reads args from
+    /// env-var `env_var_name`.
+    ///
+    /// Defaults to read from "VAR" if `env_var_name` is `None`.
+    pub fn get_matches_with_env(self, env_var_name: Option<String>)
```

Comment:
Let's change this function declaration to:

`pub fn get_matches_with_env<S: Into<&'b str>(self, env_var_name: S) -> ArgMatches<'a> {`

This will allow people to use string slices too and not have to allocate a string if they don't need to. This will also require a few lines to change inside the function too, but not many.